app/pages/shared_documents/shared_documents.php

```diff
@@ -0,0 +1,114 @@
+<?php
+restrict_access();
+
+$can_edit = check_auth(Access::$ADD_EVENTS);
+
+# public files
+$shared_files_users = em()->getRepository(SharedFile::class)->findBy(['permission_level' => Permission::USER]);
+
+# upper authorisation files
+if ($can_edit) {
+    $shared_files_coach_staff = em()->getRepository(SharedFile::class)->findBy(['permission_level' => Access::$ADD_EVENTS]);
```

Comment:
Does this work? It feels like you could achieve something that might be more efficient with DQL and `WHERE ... IN ...`

Reply:
Looks like it's working. You want me to do some benchmarking on that ?